src/models/streaming/BufferedDuplex.ts

```diff
@@ -0,0 +1,43 @@
+import { Duplex, PassThrough } from 'stream';
+
+export default class BufferedDuplex extends Duplex {
+    input: PassThrough;
+    output: PassThrough;
+    areOutputHandlersSetup: boolean;
+    constructor (input: PassThrough, output: PassThrough, options?: any) {
+        super(options);
+        this.input = input;
+        this.output = output;
+        this.areOutputHandlersSetup = false;
+    }
+
+    _write(chunk: any, encoding: any, callback: any): boolean {
+        return this.input.write(chunk, encoding, callback);
+    }
+
+    _read(size: any): any {
+        if (!this.areOutputHandlersSetup){
+            return this.setUpOutputHandlersAndRead(size);
+        }
+        return this.readOutput(size);
+    }
+
+    private setUpOutputHandlersAndRead(size : any): void {
+        var self = this;
+        self.output
+            .on('readable', function () {
```

Comment:
I believe using `() =>` arrow functions preserves `this` through the calls, so you don't have to do `var self = this` for when you want to pass `this` through to callbacks. 

https://hackernoon.com/javascript-es6-arrow-functions-and-lexical-this-f2a3e2a5e8c4

I also think this is standard typescript practice for us, so I would update to use `() =>` as much as possible

Reply:
Kenny, is correct. Using arrow functions everywhere can help us reduce the number of scoping errors we accidentally introduce.

The mobile team has implemented an arrow function only policy and has seen a decrease in scoping issues.